packages/node/src/policy-gen/load-for-policy.js

```diff
@@ -0,0 +1,463 @@
+      })
+    }
+
+    // Report SES violations if any were found
+    if (violationsForPackage.size > 0) {
+      reportSesViolations(violationsForPackage, { log })
+    }
+
+    // Clean up the worker pool
+    workerPool.terminate()
+  }
+}
+
+/**
+ * @param {Map<CanonicalName, GlobalPolicy>} globalsForPackage
+ * @param {Map<CanonicalName, BuiltinPolicy>} builtinsForPackage
+ * @param {Map<CanonicalName, PackagePolicy>} packagesForPackage
+ * @param {LavaMoatPolicy} [policyOverride]
+ * @returns {MergedLavaMoatPolicy}
+ */
+const compilePolicy = (
+  globalsForPackage,
+  builtinsForPackage,
+  packagesForPackage,
+  policyOverride
+) => {
+  /** @type {LavaMoatPolicy} */
+  const policy = { resources: {} }
+  for (const [canonicalName, globalPolicy] of globalsForPackage) {
+    // convert from Map into plain object
+    /** @type {GlobalPolicy} */
+    const plainGlobalPolicy = {}
+    for (const [
+      key,
+      value,
+    ] of /** @type {[name: string, value: GlobalPolicyValue][]} */ (
+      entries(globalPolicy)
+    )) {
+      plainGlobalPolicy[key] = value
+    }
+    policy.resources[canonicalName] = {
+      ...policy.resources[canonicalName],
+      globals: plainGlobalPolicy,
+    }
+  }
+
+  for (const [canonicalName, builtinPolicy] of builtinsForPackage) {
+    policy.resources[canonicalName] = {
+      ...policy.resources[canonicalName],
+      builtin: builtinPolicy,
+    }
+  }
+
+  for (const [canonicalName, packagePolicy] of packagesForPackage) {
+    policy.resources[canonicalName] = {
+      ...policy.resources[canonicalName],
+      packages: packagePolicy,
+    }
+  }
+
+  const mergedPolicy = mergePolicies(policy, policyOverride)
+
+  return mergedPolicy
+}
+
+/**
+ * Creates a module source inspector function with the provided dependencies.
+ *
+ * @param {Loggerr} log
+ * @param {WorkerPool<InspectMessage, PoliciesMessage | ErrorMessage>} workerPool
+ * @param {Set<FileUrlString>} inspectedModules
+ * @param {Set<FileUrlString>} modulesToInspect
+ * @param {Map<CanonicalName, GlobalPolicy>} globalsForPackage
+ * @param {Map<CanonicalName, BuiltinPolicy>} builtinsForPackage
+ * @param {Map<CanonicalName, StructuredViolationsResult>} violationsForPackage
+ * @param {Set<Promise<FileUrlString>>} pendingInspections
+ * @returns {(
+ *   moduleSource: Omit<
+ *     LocalModuleSource,
+ *     'location' | 'sourceLocation' | 'sourceDirname' | 'parser' | 'record'
+ *   > & { language: Language; location: FileUrlString },
+ *   canonicalName: CanonicalName
+ * ) => void}
+ */
+const createModuleSourceInspector = (
+  log,
+  workerPool,
+  inspectedModules,
+  modulesToInspect,
+  globalsForPackage,
+  builtinsForPackage,
+  violationsForPackage,
+  pendingInspections
+) => {
+  let messageCount = 0
+
+  /**
+   * @param {Omit<
+   *   LocalModuleSource,
+   *   'location' | 'sourceLocation' | 'sourceDirname' | 'parser' | 'record'
+   * > & { language: Language; location: FileUrlString }} moduleSource
```

Comment:
Is this an artifact of waiting for latest Endo, or does the type really differ frmo the value we're getting here? I f so, why is it rewired here and not defined somewhere as a new type for our internal use? I'd like to get the origin story of what's going on here.

Reply:
It's the `ModuleSourceForModuleSourceHook` type, which didn't exist at one point.  Now it does and I can use it.